src/info/mod.rs

```diff
@@ -364,4 +372,85 @@ mod tests {
         );
         Ok(())
     }
+
+    type Result<T = ()> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+    #[test]
+    fn test_bare_repo() -> Result {
+        let repo_path = git_testtools::scripted_fixture_repo_read_only("bare_repo.sh").unwrap();
+        let safe_repo =
+            ThreadSafeRepository::open_opts(repo_path, open::Options::isolated()).unwrap();
+        let repo = safe_repo.to_thread_local();
+        let res = Info::init_repo_path(&repo);
+        assert!(res.is_err(), "oops, info was returned on a bare git repo");
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            "please run onefetch inside of a non-bare git repository"
+        );
+        Ok(())
+    }
+
+    fn test_language_repo_again(config: &Config, re: &Regex) {
+        let info = Info::new(config).unwrap();
+        let info_str = format!("{}", info);
+        let info_u8 = strip_ansi_escapes::strip(&info_str).unwrap();
+        let simple_info_str = std::str::from_utf8(&info_u8).unwrap();
+        assert!(
+            re.is_match(&simple_info_str),
+            "OOPS, REGEX\n{}\nDOESNT MATCH\n{}",
+            re.to_string(),
+            simple_info_str
+        );
+    }
+
+    #[test]
+    fn test_language_repo() -> Result {
+        let repo_path = git_testtools::scripted_fixture_repo_read_only_with_args(
+            "language_repo.sh",
+            ["verilog"],
+        )
+        .unwrap();
+        let safe_repo =
+            ThreadSafeRepository::open_opts(repo_path.join("verilog"), open::Options::isolated())?;
+        let repo = safe_repo.to_thread_local();
+
+        // TEST JSON SERILIZER FIRST
+        let mut config: Config = Config {
+            input: repo.path().to_path_buf(),
+            ..Default::default()
+        };
+        let info = Info::new(&config).unwrap();
+        assert_json_snapshot!(
+            serde_json::to_value(info).unwrap(),
```

Comment:
```suggestion
            info,
```

The info struct already implements `Serialize`

Reply:
see [69c6229](https://github.com/o2sh/onefetch/pull/827/commits/69c62296af280e0b7bc05ea2eb00ea2564e0ff92)